app/controllers/concerns/authentication.rb

```diff
@@ -18,6 +18,7 @@ def require_authentication
 
     return if current_user?
 
+    session[:last_course_id] = params[:course_id] if params.key?(:course_id)
```

Comment:
As discussed, it's worth adding a code comment here to explain that this will only remember the `course_id` param for routes that a) are authenticated, and b) contain a `course_id` param. For example, the new/edit subscription route.